internal/storage/multi_store.go

```diff
@@ -0,0 +1,333 @@
+package storage
+
+import (
+	"fmt"
+
+	ctypes "github.com/containers/image/v5/types"
+	cstorage "github.com/containers/storage"
+	"github.com/sirupsen/logrus"
+)
+
+// multiStoreServer contains the store image servers based on the configured storage drivers
+type multiStoreServer struct {
+	store      map[string]ImageServer
+	multiStore MultiStore
+}
+
+type MultiStoreServer interface {
+	GetDefaultStorage() cstorage.Store
+	GetAllStores() []cstorage.Store
+	GetImageServer(driver string) ImageServer
+	GetStoreForContainer(idOrName string) (cstorage.Store, error)
+	GetStoreForImage(imageID string) (cstorage.Store, error)
+	FromContainerDirectory(id, file string) ([]byte, error)
+	ContainerRunDirectory(id string) (string, error)
+	ContainerDirectory(id string) (string, error)
+	Shutdown(force bool) (layers []string, err error)
+	GraphRoot() string
+	GetImageServerForImage(image string) (ImageServer, error)
+	GetStore() MultiStore
+	ListAllImages(ctx *ctypes.SystemContext, filter string) ([]ImageResult, error)
+	ResolveNames(systemContext *ctypes.SystemContext, imageName string) ([]string, error)
+	ImageStatus(systemContext *ctypes.SystemContext, filter string) (*ImageResult, error)
+}
+
+func NewMultiStoreServer(store map[string]ImageServer, istore MultiStore) MultiStoreServer {
+	return &multiStoreServer{
+		store:      store,
+		multiStore: istore,
+	}
+}
+
+func (s *multiStoreServer) GetDefaultStorage() cstorage.Store {
+	return s.multiStore.GetDefaultStorage()
+}
+
+// GetImageServer returns the ImageServer for the selected storage driver. If the driver is empty or not found, the default storage driver is returned
+func (s *multiStoreServer) GetImageServer(driver string) ImageServer {
+	iserver, ok := s.store[driver]
+	if !ok {
+		v := s.store[s.multiStore.GetDefaultStorageDriver()]
+		return v
+	}
```

Comment:
I'm not sure if we should error here, at least if the not found case which should be impossible.